src/mono/System.Private.CoreLib/src/System/Reflection/ModifiedType.Mono.cs

```diff
@@ -7,14 +7,26 @@ internal partial class ModifiedType
     {
         internal struct TypeSignature
         {
+            internal readonly RuntimeType SignatureHolder;
+            internal int ParameterIndex;
+            public TypeSignature(RuntimeType signatureHolder, int parameterIndex)
+                => (SignatureHolder, ParameterIndex) = (signatureHolder, parameterIndex);
+            // TODO: MethodInfo
```

Comment:
This is still experimental, but the basic idea is to avoid lifting up `MonoMethodSignature` to managed side, and instead use the RuntimeType and icalls to get required information.